### ISSUE TITLE
Setup CI/CD For supported platforms

### ISSUE THREAD

**Issue by vonerval** [arrayio/array-io-keychain#11]
Setup CI/CD (Jenkins) For supported platforms:
- Linux
- OS X
- Windows

**Comment by lavrenov** [comment]
Upd. Status: Waiting developers getting ready for CI.

**Comment by roman-modelist-dev** [comment]
array-io-core repo has been added into array-io-keychain as submodule.
Also I configured cmake to automaticaly build array-io-core dependencies and link it into array-io-keychain binaries.

Finally:
To build array-io-keychain on linux and macos it need:
 - to git clone with submodules array-io-keychain repo
 - run cmake in build directory with -G "Unix Makefiles" (it may need to define BOOST_ROOT and OPENSSL_ROOT_DIR variables to specify dependencies locations)

To build array-io-keychain on windows it need:
 - to git clone with submodules array-io-keychain repo
 - run cmake in build directory with -G "Visual Studio 15 2016 Win64" (it may need to define BOOST_ROOT and OPENSSL_ROOT_DIR variables to specify dependencies locations)
 - to change build toolchain in project settings fot keychain_common project from Visual Studio 2017 (v141) to LLVM-vs2017
 - build keychain-service_win and keychain targets in solution

Unfortunately, I don't know how to set LLVM-vs2017 toolchain for certain project for Visual Studio cmake generator. It need some research. It may be some specific cmake variable. Or we can just modify vcxproj  files of keychain_common project by bat script after project generation step.
